fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -233,14 +231,7 @@ class HomeFragment : Fragment() {
         val currentWallpaperName = requireContext().settings().currentWallpaperName
         applyWallpaper(wallpaperName = currentWallpaperName, orientationChange = false)
 
-        currentMode = CurrentMode(
-            requireContext(),
-            requireComponents.fenixOnboarding,
-            browsingModeManager,
-            ::dispatchModeChanges,
-        )
-
-        components.appStore.dispatch(AppAction.ModeChange(currentMode.getCurrentMode()))
+        components.appStore.dispatch(AppAction.ModeChange(Mode.fromBrowsingMode(browsingModeManager.mode)))
```

Comment:
I'm not able to easily follow this line. Could we get a quick comment for what's happening here?

Reply:
This follows the same logic as https://github.com/mozilla-mobile/firefox-android/pull/1484#discussion_r1203127927 where in the Home fragment when there is no onboarding mode, all we are trying to determine is whether or not we're in the normal or private mode.